src/cmd.rs

```diff
@@ -71,6 +71,59 @@ impl<'a, T: FromRedisValue> Iterator for Iter<'a, T> {
     }
 }
 
+use crate::aio::ConnectionLike as AsyncConnection;
+
+/// Represents a redis iterator that can be used with async connections.  
+#[cfg(feature = "aio")]
+pub struct AsyncIter<'a, T: FromRedisValue + 'a> {
+    batch: Vec<T>,
+    cursor: u64,
+    con: &'a mut (dyn AsyncConnection + 'a),
+    cmd: Cmd,
+}
+
+#[cfg(feature = "aio")]
+impl<'a, T: FromRedisValue + 'a> AsyncIter<'a, T> {
+    /// A [futures::Stream](https://docs.rs/futures/0.3.3/futures/stream/trait.Stream.html)
+    /// over values returned from the redis cursor.
+    /// ```rust,no_run
+    /// let iter = conn.sscan("set_key").await.unwrap();
+    /// let stream = iter.stream();
+    /// futures::pin_mut!(stream);
+    /// while let Some(element) = stream.next().await {
+    ///    //Do something with element
+    /// }
+    /// ```
+    #[inline]
+    pub fn stream(self) -> impl futures::Stream<Item = T> + 'a {
+        // we need to do this in a loop until we produce at least one item
+        // or we find the actual end of the iteration.  This is necessary
+        // because with filtering an iterator it is possible that a whole
+        // chunk is not matching the pattern and thus yielding empty results.
+        futures::stream::unfold(self, |mut cur_self| async move {
```

Comment:
Could extract this async block to a `next` function on `AsyncIter`. Then the `futures` dependency is a bit more "opt-in".

Reply:
Done.